plugins/modules/gitlab_merge_request.py

```diff
@@ -0,0 +1,419 @@
+#!/usr/bin/python
+# -*- coding: utf-8 -*-
+
+# Copyright (c) 2023, Ondrej Zvara (ozvara1@gmail.com)
+# Based on code:
+# Copyright (c) 2021, Lennert Mertens (lennert@nubera.be)
+# Copyright (c) 2021, Werner Dijkerman (ikben@werner-dijkerman.nl)
+# Copyright (c) 2015, Werner Dijkerman (ikben@werner-dijkerman.nl)
+# Copyright (c) 2019, Guillaume Martinez (lunik@tiwabbit.fr)
+# GNU General Public License v3.0+ (see LICENSES/GPL-3.0-or-later.txt or https://www.gnu.org/licenses/gpl-3.0.txt)
+# SPDX-License-Identifier: GPL-3.0-or-later
+from __future__ import absolute_import, division, print_function
+__metaclass__ = type
+
+DOCUMENTATION = '''
+module: gitlab_merge_request
+short_description: Create, update or delete a merge requests.
+version_added: 4.2.0
+description:
+  - This module allows to create, update or merge requests.
+author:
+  - zvaraondrej (@zvaraondrej)
+requirements:
+  - python >= 2.7
+  - python-gitlab >= 2.3.0
+extends_documentation_fragment:
+  - community.general.auth_basic
+  - community.general.gitlab
+  - community.general.attributes
+
+attributes:
+  check_mode:
+    support: full
+  diff_mode:
+    support: none
+
+options:
+  state:
+    description:
+      - Create or delete merge request.
+    default: present
+    type: str
+    choices: ["present", "absent"]
+  project:
+    description:
+      - The path or name of the project.
+    required: true
+    type: str
+  source_branch:
+    description:
+      - Merge request's source branch.
+      - Ignored while updating existing merge request.
+    required: true
+    type: str
+  target_branch:
+    description:
+      - Merge request's target branch.
+    required: true
+    type: str
+  title:
+    description:
+      - A title for the merge request.
+    type: str
+    required: true
+  description:
+    description:
+      - A description for the merge request.
+      - Gets overriden by a content of file specified at description_path, if found.
+    type: str
+  description_path:
+    description:
+      - A path of file containing merge request's description.
+      - Accepts markdown formatted files.
+    type: str
+  labels:
+    description:
+      - Comma separated list of label names.
+    type: str
+    default: ""
+  remove_source_branch:
+    description:
+      - Flag indicating if a merge request should remove the source branch when merging.
+    type: bool
+    default: false
+  state_filter:
+    description:
+      - Filter specifying state of merge requests while searching.
+      - If empty string, merge requests of all states are searched.
+      - Possible values are "opened", "closed", "locked", "merged" or empty string.
```

Comment:
An empty string is obviously not allowed, since `choices` does not list it. In any case, since `choices` is there, the values shouldn't be listed another time (except if there is more information on them).